cpp/src_prims/selection/knn.cuh

```diff
@@ -471,8 +471,12 @@ void class_probs(std::vector<float *> &out, const int64_t *knn_indices,
      * knn_indices and labels
      */
     device_buffer<int> y_normalized(allocator, stream, n_index_rows);
-    MLCommon::Label::make_monotonic(y_normalized.data(), y[i], n_index_rows,
-                                    stream, allocator);
+    device_buffer<int> y_tmp(allocator, stream, n_index_rows + n_unique_labels);
+    updateDevice(y_tmp.data(), y[i], n_index_rows, stream);
+    updateDevice(y_tmp.data() + n_index_rows, uniq_labels[i], n_unique_labels,
+                 stream);
+    MLCommon::Label::make_monotonic(y_normalized.data(), y_tmp.data(),
```

Comment:
When having lots of very small partitions or a very small batch size, the labels array that is processed may not contain every of the unique labels. Because of this the `make_monotonic` goes haywire and subsequent probabilities array for each of the unique classes is wrong.

Reply:
Did you get a chance to file this issue, @viclafargue ?

Reply:
Yes, it's #2869 .